### PR TITLE
OpenSearch multi-tenant vector store sample

### PR DESCRIPTION
**Pull request by nihilg**
*Issue #, if available:*

*Description of changes:*
OpenSearch multi-tenant vector store sample

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/multi-tenant-vector-database/amazon-opensearch/cfn/opensearch-template.yaml [hunk 1]
```diff
@@ -0,0 +1,959 @@
...
+
+                  parameters = event['ResourceProperties']
+                  if 'Bucket' not in parameters or len(parameters['Bucket']) < 1:
+                      logger.error("Bucket parameter not provided")
+                      if respond_to_cfn:
+                          send_cfn_response(event, context, "FAILED", response_data)
+                      return
+
+                  #
+                  # Do stuff
```
Comment (jhart0) [inline samples/multi-tenant-vector-database/amazon-opensearch/cfn/opensearch-template.yaml:562]: Can this comment be removed or explain what this method does?

File: samples/multi-tenant-vector-database/amazon-opensearch/cfn/opensearch-template.yaml [hunk 2]
```diff
@@ -0,0 +1,959 @@
...
+        SecurityGroupIds:
+          - !Ref OpenSearchSecurityGroup
+      AccessPolicies:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS:
+                - "*"
+                #- !GetAtt OpenSearchAdminRole.Arn
```
Comment (jhart0) [inline samples/multi-tenant-vector-database/amazon-opensearch/cfn/opensearch-template.yaml:354]: Remove or add comment to justify

**Review by jhart0 — CHANGES_REQUESTED** [review]
Looks great! Just a couple of small notes on some of the comments in there. Also, can we add a description to the root README of the project describing this sample please?

File: samples/multi-tenant-vector-database/amazon-opensearch/cfn/opensearch-template.yaml (hunk 1 above)
Comment (nihilg) [reply samples/multi-tenant-vector-database/amazon-opensearch/cfn/opensearch-template.yaml]: Removed !

File: samples/multi-tenant-vector-database/amazon-opensearch/cfn/opensearch-template.yaml (hunk 2 above)
Comment (nihilg) [reply samples/multi-tenant-vector-database/amazon-opensearch/cfn/opensearch-template.yaml]: Removed !!

**Review by jhart0 — APPROVED** [review]

**Pull request merged**